modules/ipc/httpCompatSocket.js

```diff
@@ -0,0 +1,74 @@
+
+const util = require('util');
+const EventEmitter = require('events').EventEmitter;
+const http = require('http');
+
+function call(hostname, port, data) {
+    return new Promise((resolve, reject) => {
+        console.log(data);
+        const req = http.request({
+            hostname: hostname,
+            port: port,
+            path: '/',
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+                'Content-Length': data.length
+            }
+        }, resolve);
+
+        // TODO Not sure why it's needed
+        // Seems that responses hang somewhere without it
+        setTimeout(() => {
+            req.on('error', reject);
+            req.write(data);
+            req.end();
+        }, 50);
+    });
+}
+
+function HttpCompatSocket(hostname, port) {
+    this.call = call.bind(null, hostname, port);
+}
+
+util.inherits(HttpCompatSocket, EventEmitter);
+
+Object.assign(HttpCompatSocket.prototype, {
+    call: null,
+    encoding: 'utf8',
+    writable: true, // socket is always writable
+
+    connect(event) {
+        this.call(JSON.stringify({
+            jsonrpc: "2.0",
+            id: 0,
+            method: "eth_syncing",
```

Comment:
better use `eth_accounts` for that, as `eth_syncing` is not available in older nodes.